### PR TITLE
Updates including support for building VMware images

### PR DESCRIPTION
**Pull request by yankcrime**
Headline changes in this PR are:
* Add support for creating an image suitable for use under VMware
* Bring Ansible up-to-date and refactor some of the code as a result
* Update Docker and other supporting tools.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -42,7 +34,7 @@ Pre-provisioning this box front-loads internet access requirements, flakiness, a
   - Default Vagrant SSH key
   - Configure SSH daemon (disable DNS and API authn for speed)
   - Set box build time (`/etc/vagrant_box_build_time`)
-- VirtualBox
+- VirtualBox and VMware
```
Comment (adamtheturtle) [inline README.md]: Is this right? do all the points below also apply to VMWare?

Also - let's call it "VMWare Fusion".

File: bin/zerodisk.sh [hunk 2]
```diff
@@ -1,15 +1,12 @@
-#!/usr/bin/env bash
-
-set -o errexit
```
Comment (adamtheturtle) [inline bin/zerodisk.sh]: Why remove these settings?

Comment (adamtheturtle) [reply bin/zerodisk.sh]: (oh they've moved to the top)

File: Vagrantfile [hunk 3]
```diff
@@ -11,16 +11,23 @@ Vagrant.configure(2) do |config|
...
-      v.cpus = 2
-      v.memory = 2048
-      v.customize ['modifyvm', :id, '--natdnshostresolver1', 'on']
-    end
+  config.vm.provider "virtualbox" do |vb|
+    vb.name = "dcos-centos-virtualbox"
+    vb.cpus = 2
+    vb.memory = 2048
+    vb.customize ['modifyvm', :id, '--natdnshostresolver1', 'on']
+    vb.linked_clone = true
```
Comment (adamtheturtle) [inline Vagrantfile]: @yankcrime Learning for me! Should I use this in minidcos?

Comment (adamtheturtle) [reply Vagrantfile]: I see from the docs that when this is used:

> Note: the generated master VMs are currently not removed automatically by Vagrant. This has to be done manually. However, a master VM can only be removed when there are no linked clones connected to it.

Is this accounted for somehow? Or will users now have to know somehow to clean up?

File: Vagrantfile [hunk 4]
```diff
@@ -11,16 +11,23 @@ Vagrant.configure(2) do |config|
...
   end
+  config.vm.provider "vmware_fusion" do |vmware|
+    vmware.vmx['memsize'] = 2048
```
Comment (adamtheturtle) [inline Vagrantfile]: I guess that the memory used in `vmware` is purposefully the same as in `vb` (and same with `cpus`. If so, can we use a variable?

File: ansible/roles/guestadditions/tasks/main.yml [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+# Configure guest additions
+
+- name: Install common packages for guest additions
+  include: install.yml
+
+- name: Install VirtualBox Guest Additions
+  include: virtualbox.yml
+  when: ansible_facts['virtualization_type'] == "VirtualBox"
+
+- name: Install VMware TOols
```
Comment (adamtheturtle) [inline ansible/roles/guestadditions/tasks/main.yml]: `s/TOols/Tools`

File: README.md (hunk 1 above)
Comment (yankcrime) [reply README.md]: The generated image _should_ work with VMware Workstation and Fusion, although I've only tested the latter.

Comment (yankcrime) [reply README.md]: You're right though and we should be clear that that's what we mean :)

File: Vagrantfile (hunk 3 above)
Comment (yankcrime) [reply Vagrantfile]: I'd recommend enabling it, yes - it does noticeably reduce startup time for new machines.  Users will have to remove the old master Vagrant box as / when they upgrade after we publish a new version.  I'll add something to the README along those lines.

**Comment by jongiddy** [comment]
Updated the CentOS box to more recent version that is available to download. Previous version has gone.

Exclude VMWare from CI build until we have a need for it.

**Pull request merged**